inc/azure_c_logging/xlogging.h

```diff
@@ -87,14 +87,22 @@ typedef void(*LOGGER_LOG_GETLASTERROR)(const char* file, const char* func, int l
 
 #define LOG(log_category, log_options, format, ...) MU_C2(LOG_,log_category)(log_category, log_options, format, ##__VA_ARGS__)
 
+#ifdef _MSC_VER
+static __declspec(thread) char stackAsString[STACK_MAX_CHARACTERS];
+static __declspec(thread) char formatWithStack[FORMAT_MAX_CHARACTERS];
+#else
+/*for C11 compilers*/
+static _Thread_local char stackAsString[STACK_MAX_CHARACTERS];
```

Comment:
>static [](start = 0, length = 6)

Will this have one stackAsString for each translation unit?
Seems like we might want to have the variable and function that uses it to barf the stack in the .c file.

Reply:
a VERY good observation.

logging absolutely needs a "rework". I've placed the variables appropriately now. 

---
In reply to: [504394400](https://github.com/Azure/azure-c-logging/pull/18#discussion_r504394400) [](ancestors = 504394400)